src/main/java/org/spongepowered/api/data/key/KeyFactory.java

```diff
@@ -114,16 +119,45 @@ public int hashCode() {
             @Override
             public String toString() {
                 if (this.string == null) {
-                    this.string = "Key{Value:" + valueToken.getRawType().getSimpleName() + "<"
-                            + elementToken.getRawType().getSimpleName() + ">, Query: "
-                            + query.toString() + "}";
+                    this.string = "Key{Value:" + valueToken.toString() + ",Query:" + query.toString() + "}";
                 }
-
                 return this.string;
             }
         };
     }
 
+    public static <E> Key<Value<E>> makeValueKey(TypeToken<E> elementToken,
+            DataQuery query, String id, String name) {
+        final TypeToken<Value<E>> valueToken = new TypeToken<Value<E>>() {}
+                .where(new TypeParameter<E>() {}, elementToken);
+        return makeSingleKey(elementToken, valueToken, query, id, name);
+    }
+
```

Comment:
Need some java docs to differentiate what the different variations of the methods are.